### ISSUE TITLE
Setters are not being called in `new`

### ISSUE THREAD

**Issue by drujensen** [amberframework/granite#381]
When calling `new` with a list of params, it should call the setters for a column so someone can override the default behavior.

**Comment by Blacksmoke16** [comment]
Related to #370.  This wouldn't be an issue if we stopped creating initializers behind the scenes.  Also for whatever its worth, I don't think Granite ever worked like that, as the old [cast_to_field](https://github.com/amberframework/granite/blob/v0.16.0/src/granite/fields.cr#L138) was setting the ivars directly as well.

**Comment by drujensen** [comment]
Yeah, I agree.  We should treat `params` as a serializer, similar to `json`, `yaml`, or `xml`.  In Amber, we could initialize the object using `Class.from_params` method.

Question: should serializers call the setter method?

**Comment by Blacksmoke16** [comment]
Or treat it more like `DB::ResultSet` and overload `#new` for `HTTP::Params` or something like that.

It would be doable to make it configurable.  The serializer we use at work (in PHP) allows you to [control](https://jmsyst.com/libs/serializer/master/reference/annotations#accesstype) the access method; whether it should use reflection or public methods.  We could do something similar as I can see the pros and cons of each approach.
